runtime/near-vm-runner/tests/test_error_cases.rs

```diff
@@ -420,12 +476,58 @@ fn bad_many_imports() -> Vec<u8> {
 
 #[test]
 fn test_bad_many_imports() {
-    let result = make_simple_contract_call(&bad_many_imports(), b"hello");
-    assert_eq!(result.0, Some(vm_outcome_with_gas(0)));
-    if let Some(VMError::FunctionCallError(FunctionCallError::LinkError { msg })) = result.1 {
-        eprintln!("{}", msg);
-        assert!(msg.len() < 1000, format!("Huge error message: {}", msg.len()));
-    } else {
-        panic!(result.1);
-    }
+    with_vm_variants(|vm_kind: VMKind| {
+        let result = make_simple_contract_call_vm(&bad_many_imports(), b"hello", vm_kind);
+        assert_eq!(result.0, Some(vm_outcome_with_gas(0)));
+        if let Some(VMError::FunctionCallError(FunctionCallError::LinkError { msg })) = result.1 {
+            eprintln!("{}", msg);
+            assert!(msg.len() < 1000, format!("Huge error message: {}", msg.len()));
+        } else {
+            panic!(result.1);
+        }
+    });
+}
+
+fn external_call_contract() -> Vec<u8> {
+    wabt::wat2wasm(
+        r#"
+            (module
+              (import "env" "prepaid_gas" (func (;0;) (result i64)))
+              (export "hello" (func 1))
+              (func (;1;)
+                  (drop (call 0))
+                  )
+            )"#,
+    )
+    .unwrap()
+}
+
+#[test]
+fn test_external_call_ok() {
+    with_vm_variants(|vm_kind: VMKind| {
+        assert_eq!(
+            make_simple_contract_call_vm(&external_call_contract(), b"hello", vm_kind),
+            (Some(vm_outcome_with_gas(272974500)), None)
+        );
+    });
+}
+
+#[test]
+fn test_external_call_error() {
+    with_vm_variants(|vm_kind: VMKind| {
+        assert_eq!(
+            make_simple_contract_call_with_gas_vm(
+                &external_call_contract(),
+                b"hello",
+                100,
+                vm_kind
+            ),
+            (
+                Some(vm_outcome_with_gas(100)),
+                Some(VMError::FunctionCallError(FunctionCallError::HostError(
+                    HostError::GasExceeded
+                )))
+            )
+        );
+    });
```

Comment:
It would add safety if tests here were testing for all variants of `FunctionCallError` enum. It might go into a different PR.

Reply:
Agree, we generally need to think about better coverage.